scripts/compare_snapshots.py

```diff
@@ -0,0 +1,92 @@
+import json
+import sys
+import logging
+
+log = logging.getLogger(__name__)
+
+
+def do_values_match(a, b, file_name_a, file_name_b):
+    """Compare a and b recursively, logging info about any mismatch found
+    """
+    aType = type(a)
+    bType = type(b)
+    if aType != bType:
+        log.warning("Mismatch: Values have different types: %s vs %s", aType, bType)
+        return False
+
+    if aType == dict:
+        if len(a) != len(b):
+            log.warning(
+                "Mismatch: Dicts have different lengths (%s vs %s). a_keys=%s vs b_keys=%s",
+                len(a),
+                len(b),
+                a.keys(),
+                b.keys(),
+            )
+            return False
+        for key in a:
+            a_value = a[key]
+            b_value = b.get(key)
+
+            if b_value is None:
+                log.warning(
+                    "Mismatch: Key %s present in %s but missing from %s dict with keys %s",
+                    key,
+                    file_name_a,
+                    file_name_b,
+                    b.keys(),
+                )
+                return False
+
+            if not do_values_match(a_value, b_value, file_name_a, file_name_b):
+                log.warning("Mismatch: Dict values at key %s do not match", key)
```

Comment:
Maybe we could log the values here so the dev can quickly identify and fix the issue at a glance

Reply:
The values should get logged in the recursive call to do_values_match on 41 resulting in output like

```
Performing diff of test_extension_snapshot.json against correct_extension_snapshot.json
Values AWS::Lambda::LayerVersion and AWS::Lambda::LayerVersion-BROKEN do not match
Mismatch: Dict values at key Type do not match
Mismatch: Dict values at key ProviderLevelLayerLambdaLayer do not match
Mismatch: Dict values at key Resources do not match
FAILURE: There were differences between the test_extension_snapshot.json and correct_extension_snapshot.json. Review the diff output above.
If you expected the test_extension_snapshot.json to be different generate new snapshots by running this command from a development branch on your local repository: 'UPDATE_SNAPSHOTS=true ./scripts/run_integration_tests.sh'
```

Reply:
Ah totally didn't notice this, nice work!